### PR TITLE
boostrap: improve testing and support v23.1

### PR DESCRIPTION
**Pull request by postamar**
This commit adds a test in the bootstrap package which checks that initial values can be generated for each and every release which is supported by the current version.

We expect this test, TestSupportedReleases, to fail loudly the next time a release engineer bumps the major or minor version counter. While getting the test to pass does increase the burden of the release engineer, doing so should be a very mechanical process. It involves copy-pasting the contents of the test data file for the data-driven TestInitialValuesToString test in the new release branch.

This commit therefore also adds the hard-coded values for the 23.1 release which was until now not supported by the bootstrap package.

Informs #103956.

Release note: None.

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
:lgtm: Also, good idea with the test it will force similar version bumps to have the hardcoded values added!

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/104336#-:-NXGcbHt0sRk7FnINps4:b28wlom)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @healthy-pod and @renatolabs)

**Comment by postamar** [comment]
Thanks for the review!

bors r+

File: pkg/sql/catalog/bootstrap/initial_values.go [hunk 1]
```diff
@@ -393,4 +356,298 @@ const (
 ]`
 	// nonSystem222SHA256 is the SHA-256 hash of the above constant.
 	nonSystem222SHA256 = `49b3e2d0bb0d18d711d2fff94b0c11db2e5caecf3c4fcc27d9bd277694647afd`
+
+	// system222 is the hard-coded string representation of the
```
Comment (renatolabs) [inline pkg/sql/catalog/bootstrap/initial_values.go:360]: `system231`

File: pkg/sql/catalog/bootstrap/initial_values.go [hunk 2]
```diff
@@ -393,4 +356,298 @@ const (
...
+	// initial values for a system tenant as produced by the 23.1
+	// release binary.
+	//
+	// This string literal was generated by copying over the expected results
+	// for the TestInitialValuesToString test in the release-23.1 branch.
+	system231 = `[{"key":"04646573632d696467656e","value":"01c801"}
```
Comment (renatolabs) [inline pkg/sql/catalog/bootstrap/initial_values.go:366]: We could consider using `embed` to make the actual source code file lighter, especially with LTS and us having to support multiple releases.

**Review by renatolabs — COMMENTED** [review]
Very late to the party, just some notes for possible future work. Thanks for the new test!

**Pull request merged**